src/core/lib/surface/server.cc

```diff
@@ -730,11 +730,10 @@ static void server_on_recv_initial_metadata(void* ptr, grpc_error* error) {
   if (calld->host_set && calld->path_set) {
     /* do nothing */
   } else {
-    grpc_error* src_error = error;
-    error = GRPC_ERROR_CREATE_REFERENCING_FROM_STATIC_STRING(
-        "Missing :authority or :path", &error, 1);
-    /* Pass the src_error reference to calld->recv_initial_metadata_error */
-    calld->recv_initial_metadata_error = src_error;
+    /* Pass the error reference to calld->recv_initial_metadata_error */
+    calld->recv_initial_metadata_error =
+        GRPC_ERROR_CREATE_REFERENCING_FROM_STATIC_STRING(
```

Comment:
I think we still need to reset `error` here, since it will be used on line 746 below.